ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFArrayPosition.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hive.ql.udf.generic;
+
+import org.apache.hadoop.hive.ql.exec.Description;
+import org.apache.hadoop.hive.ql.exec.UDFArgumentException;
+import org.apache.hadoop.hive.ql.metadata.HiveException;
+import org.apache.hadoop.hive.serde2.objectinspector.ListObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.ObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.primitive.PrimitiveObjectInspectorFactory;
+import org.apache.hadoop.io.IntWritable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * GenericUDFArrayPosition.
+ */
+@Description(name = "array_position", value = "_FUNC_(array, element) - Returns the position of the first occurrence of "
+    + "element in array. Array indexing starts at 1. If the element value is NULL, a NULL is returned.", extended =
+    "Example:\n" + "  > SELECT _FUNC_(array(1, 2, 3,4,2), 2) FROM src;\n" + "  2")
```

Comment:
I originally thought arrays of Hive, Spark, MySQL, etc. were 0-based. But I also see `array_position` of Spark is 1-based...
- https://spark.apache.org/docs/latest/api/sql/index.html#array_position

```
spark-sql (default)> SELECT array(1, 2, 3, 4, 5)[3];
4
spark-sql (default)> SELECT array_position(array(1, 2, 3, 4, 5), 3);
3
```

Reply:
Yes, tried to replicate spark's behaviour